examples/tutorials/color-figure.py

```diff
@@ -0,0 +1,95 @@
+# figure based upon the z-values of the data. By default, it plots the map with the
+# equidistant cylindrical projection and with no frame.
+
+fig = pygmt.Figure()
+fig.grdimage(grid=grid)
+fig.show()
+
+########################################################################################
+# A specific color palette can be set using the optional ``cmap`` argument for
+# :meth:`pygmt.Figure.grdimage`. By default, the color palette is set to *turbo*.
+# In the example below, the color palette is set to *geo*.
+# The full list of color palette tables can be found at :gmt-docs:`cookbook/cpts.html`.
+
+fig = pygmt.Figure()
+fig.grdimage(grid=grid, cmap="geo")
+fig.show()
+
+########################################################################################
+# :meth:`pygmt.Figure.grdimage` accepts additional parameters, including  ``frame`` and
+# ``projection``.
+
+fig = pygmt.Figure()
+fig.grdimage(grid=grid, frame=True, projection="M6i", cmap="geo")
+fig.show()
+
+########################################################################################
+# The :meth:`pygmt.Figure.colorbar` method can be used to add a color bar to the figure.
+# By default, it applies the color palette created by :meth:`pygmt.Figure.grdimage`.
+#
+# The `frame` argument can be used to set the color bar labels and intervals.
+# In the example below, ``p3000`` sets the color bar tick interval to 3,000 meters,
+# and ``x+lElevation`` and ``y+lm`` set the x- and y-axis labels for the color bar.
+
+fig = pygmt.Figure()
+fig.grdimage(grid=grid, frame=True, projection="M6i", cmap="geo")
+fig.colorbar(frame=["p3000", "x+lElevation", "y+lm"])
+fig.show()
+
+########################################################################################
+# Create a custom color palette
+# -----------------------------
+#
+# The :meth:`pygmt.makecpt` method provides the option to create a custom color palette
+# for a figure. The ``cmap`` argument sets the master color palette to base the
+# new color palette on. The ``series`` argument sets the minimum and maximum values, and
+# optionally the intervals between them; without it, the new color palette defaults to
+# the min/max values of the master color palette.  The ``series`` argument can be given
+# as a string (as below) or as a list ``series=[-8500, 2500, 1000]``. Any values that
+# fall outside the range of the series will be plotted as black (lower than the minimum)
+# or white (higher than the maximum).
+#
+# The :meth:`pygmt.makecpt` method includes an ``output`` parameter, which allows the
+# custom color palette to be saved in a .cpt file. It's default value is False, which
+# uses the custom color palette as the default color palette for
+# the figure and color bar.
+
+fig = pygmt.Figure()
+pygmt.makecpt(cmap="jet", series="-9000/3000", continuous=True)
+fig.grdimage(grid=grid, projection="M6i", frame=True)
+fig.colorbar(frame=["p3000", "x+lElevation", "y+lm"])
+fig.show()
```

Comment:
It's really a bad idea to plot earth relief data using a regular CPT like `jet`. Perhaps we should use another grid data for this tutorial, for example, https://matplotlib.org/gallery/images_contours_and_fields/shading_example.html.

Reply:
@seisman My apologies, but I've been out of geoscience since 2012 and I'm pretty unfamiliar with using GMT for anything other than plotting Earth topography/bathymetry data. I chose a high-contrast CPT like `jet` because I figured any changed to the color palette would be visually apparent in the tutorial. I'm happy to change to another data set; are you suggesting following the HSV example from the gallery you posted?

Reply:
@seisman I decided to remove the custom CPT from this tutorial. I think it would make more sense to demonstrate the `makecpt` options with multiple color bars, as is done in the [GMT Tutorial](https://docs.generic-mapping-tools.org/6.1/tutorial/session-4.html).

Reply:
@willschlitzer Sorry for the late reply.

It's very common to plot earth relief as basemaps. PyGMT provides the `pygmt.datasets.load_earth_relief` function to load earth relief data into memory. However, there is no tutorial for this function yet. I feel that you can slightly modify this PR into a tutorial about "Plotting Earth relief on maps".

It would be better if you can do the following things in this tutorial:

1. loading a low-resolution grid (`01d`, or `30m`) and plot a global relief
2. try using different CPTs (geo, relief, etopo1, ...)
3. Add a colorbar
4. loading a high-resolution grid (e.g., `05m`) for a smaller region (just like what you're doing).

Please let me know what you think.

Reply:
Sure thing! I'll make the changes over the next day or two